packages/rum-core/src/domain/rumEventsCollection/view/viewMetrics/trackCommonViewMetrics.ts

```diff
@@ -86,7 +91,9 @@ export function trackCommonViewMetrics(
     },
     setLoadEvent,
     getCommonViewMetrics: () => {
-      commonViewMetrics.interactionToNextPaint = getInteractionToNextPaint()
+      const { interactionToNextPaint, interactionToNextPaintTargetSelector } = getInteractionToNextPaint() || {}
+      commonViewMetrics.interactionToNextPaint = interactionToNextPaint
+      commonViewMetrics.interactionToNextPaintTargetSelector = interactionToNextPaintTargetSelector
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏what about not flattening the intermediate data structure?
it would avoid to update the intermediary layer at each time we want to add extra properties to an already collected metrics